core/domain/email_manager.py

```diff
@@ -255,3 +297,57 @@ def send_moderator_action_email(
     _send_email(
         recipient_id, sender_id, intent, email_subject, full_email_content,
         bcc_admin=True)
+
+def send_exploration_membership_email(
+        sender_id, recipient_id, recipient_role, exploration_id,
+        exploration_title):
+    """Sends a email when a new user is given activity rights (Manager, Editor,
+    Viewer) to an exploration by creator of exploration.
+
+    Email will only be sent if recipient wants to receive these emails (i.e.
+    'can_receive_editor_role_email' is set True in recipent's preferences).
+    """
+    # Return from here if sending email is turned off.
+    if not feconf.CAN_SEND_EMAILS_TO_USERS:
+        log_new_error('This app cannot send emails to users.')
+        return
+
+    recipient_user_setting = user_services.get_user_settings(recipient_id)
+    sender_user_settings = user_services.get_user_settings(sender_id)
+    recipient_preferences = user_services.get_email_preferences(recipient_id)
+    if recipient_preferences['can_receive_editor_role_email'] is False:
+        log_new_error(
+            '%s will not receive email notification of being assigned to an '
+            'exploration' % recipient_user_setting.username)
+        return
+
+    if recipient_role == rights_manager.ROLE_OWNER:
+        role = EXPLORATION_ROLE_MANAGER
+        rights = EXPLORATION_ROLE_MANAGER_RIGHTS
+
+    elif recipient_role == rights_manager.ROLE_EDITOR:
+        role = EXPLORATION_ROLE_COLLABORATOR
+        rights = EXPLORATION_ROLE_COLLABORATOR_RIGHTS
+
+    elif recipient_role == rights_manager.ROLE_VIEWER:
+        role = EXPLORATION_ROLE_PLAYTESTER
+        rights = EXPLORATION_ROLE_PLAYTESTER_RIGHTS
+
+    else:
+        raise Exception(
+            'Invalid role: %s' % recipient_role)
+
+    raw_email_subject = EDITOR_ROLE_EMAIL_CONTENT.value['subject']
+    raw_email_body = 'Hi %s,<br><br>%s<br><br>%s' % (
+        recipient_user_setting.username,
+        EDITOR_ROLE_EMAIL_CONTENT.value['html_body'],
+        EMAIL_FOOTER.value)
+
+    email_subject = raw_email_subject % (sender_user_settings.username)
+    email_body = raw_email_body % (
+        sender_user_settings.username, exploration_title, role, rights,
+        exploration_id, exploration_title)
+
+    _send_email(
+        recipient_id, feconf.SYSTEM_COMMITTER_ID,
```

Comment:
Note that this confirms that sender_id, sender_user_settings are misnamed -- the sender is feconf.SYSTEM_COMMITTER_ID. Maybe change sender to inviter throughout in order to avoid confusion.